main/api/dependencies/auth.py

```diff
@@ -34,7 +36,16 @@ async def require_authenticated_user(
 
 
 async def require_permission_on_category(
-    user: UserModel = Depends(require_authenticated_user), category: CategoryModel = Depends(get_category_by_id)
+    user: UserModel = Depends(require_authenticated_user),
+    category: CategoryModel = Depends(get_category_by_id),
 ) -> None:
     if category.user_id != user.id:
         raise ForbiddenException("User does not have permission to perform this action")
+
+
+async def require_permission_on_item(
+    user: UserModel = Depends(require_authenticated_user),
+    item: ItemModel = Depends(get_item_by_id),
+) -> None:
+    if item.user_id != user.id:
+        raise ForbiddenException("User does not have permission to perform this action")
```

Comment:
Create `require_ownership` method to validate the ownership for both item and category, which takes the `require_item` or `require_category` as the input and then return the corresponding entity